pkg/volume/plugins.go

```diff
@@ -121,6 +122,9 @@ type VolumeHost interface {
 	// the provided spec.  See comments on NewWrapperBuilder for more
 	// context.
 	NewWrapperCleaner(spec *Spec, podUID types.UID, mounter mount.Interface) (Cleaner, error)
+
+	//Get cloud provider from kubelet
+	GetCloudProvider() cloudprovider.Interface
```

Comment:
I am torn on this one.  Modules like GCE instantiate their own instance of
CloudProvider - I don't see much strong reason to go one way or the other,
so maybe the conservative answer is to not change the interface?

On Wed, Aug 19, 2015 at 6:42 PM, Saad Ali <notifications@github.com> wrote:

> In pkg/volume/plugins.go
> <https://github.com/kubernetes/kubernetes/pull/6689#discussion_r37488358>:
>
> > @@ -121,6 +122,9 @@ type VolumeHost interface {
> >  	// the provided spec.  See comments on NewWrapperBuilder for more
> >  	// context.
> >  	NewWrapperCleaner(spec *Spec, podUID types.UID, mounter mount.Interface) (Cleaner, error)
> > +
> > +	//Get cloud provider from kubelet
> > +	GetCloudProvider() cloudprovider.Interface
>
> The cloudprovider package already provides
> <https://github.com/kubernetes/kubernetes/blob/master/pkg/cloudprovider/plugins.go#L55>
> a GetCloudProvider(...) method that should be used instead.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/kubernetes/kubernetes/pull/6689/files#r37488358>.
>